### PR TITLE
Allow character generation menu to scale to screen size

### PR DESCRIPTION
**Pull request by Coolthulhu**
Mostly replacing hardcoded widths and heights with window-size-dependent ones.
Some minor refactoring, because one fragment was so hard to read that I had to refactor it to understand what is going on and I decided to keep the cleaner version.

Fixed a bug that didn't appear in most mods: if number of skills in game was below some constant based on menu size, the game would crash.

### CODE REVIEW

File: src/newcharacter.cpp [hunk 1]
```diff
@@ -158,6 +156,11 @@ struct points_left {
     }
 };
 
+std::string empty_line( size_t len )
```
Comment (kevingranade) [inline src/newcharacter.cpp]: Seriously? Why?

**Comment by Theundyingcode** [comment]
image

I looks dumb now in my opinion

**Comment by Night-Pryanik** [comment]
I suppose scaling is just a first step to improving the interface. We'll eventually make text scale too.

**Comment by Coolthulhu** [comment]
If scaling to some smaller number is desired, it can be changed. But having it all get crammed into a 80x25 and add scrollbars everywhere was worse than having it look kinda bad at times.
Utility and readability > aesthetics

File: src/newcharacter.cpp (hunk 1 above)
Comment (Coolthulhu) [reply src/newcharacter.cpp]: There was an empty line constant. So an empty line function fills the empty void.

Comment (kevingranade) [reply src/newcharacter.cpp]: Wraping std::string( num, ' ' ) in empty_line( num ) isn't an improvement, it's just as verbose and adds no significant clarity to the code.

**Comment by Theundyingcode** [comment]
That's fine, just wanted point the issue out.

**Pull request merged**